source/mir/ndslice/slice.d

```diff
@@ -1144,17 +1144,15 @@ struct Slice(size_t _N, _Range)
         mixin(mathIndexStrideCode);
     }
 
+    static if (hasPtrBehavior!PureRange)
     this(ref in size_t[PureN] lengths, ref in sizediff_t[PureN] strides, PureRange range)
     {
         foreach (i; Iota!(0, PureN))
             _lengths[i] = lengths[i];
         foreach (i; Iota!(0, PureN))
             _strides[i] = strides[i];
-        static if (hasPtrBehavior!PureRange)
-            _ptr = range;
-        else
-            _ptr._range = range;
 
+        _ptr = range;
```

Comment:
maybe we are talking about different things, this constructor is always called with ranges that have `PtrBehavior`, so this line:

```
_ptr._range = range;
```

is never called.